draft-ietf-rtcweb-security-arch.xml

```diff
@@ -556,6 +556,163 @@
       </section>
     </section>
 
+    <section title="SDP Identity Attribute" anchor="sec.sdp-id-attr">
+      <t>
+	    The SDP 'identity' attribute is a session-level attribute that is used
+	    by an endpoint to convey its identity assertion to its peer. The identity
+        assertion value is encoded as a <xref target="RFC4648">Base-64 string</xref>
+      </t>
+      <t>
+	    The procedures in this section are based on the assumption that the identity assertion 
+        of an endpoint is bound to the fingerprints of the endpoint. This does not preclude the 
+        definition of alternative means of binding an assertion to the identity, but such means are 
+        outside the scope of this specification.
+      </t>
+      <t>
+	    The semantics of multiple 'identity' attributes within an offer or answer are undefined.
+	    Implementations SHOULD only include a single identity attribute in an
+	    offer or answer and relying parties MAY elect to ignore all but the first
+	    identity attribute. However, a single identity attribute can convey multiple 
+	    identity assertions.
+      </t>
+      <figure>
+      <artwork type="inline"><![CDATA[
+
+Name: identity
+
+Value: identity-assertion
+
+Usage Level: session
+
+Charset Dependent: no
+
+Default Value: N/A
+
+Syntax:
+
+  identity-assertion        = identity-assertion-value 
```

Comment:
do we need some syntax to merge this ABNF into what is allowed on a= line in SDP ?

Reply:
The ABNF needs to make it clear that more than one identity-assertion-value  is allowed by the syntax as the ABNF is what restricts future extensions.

Reply:
Actually, when looking closer to the ABNF (including the previous version), I am not sure whether multiple assertions is allowed. If we want to allow it, I think we should have something like:

identity-assertion-list = identity-assertion *(SEMI identity-assertion)
identity-assertion        = identity-assertion-value *(SP identity-extension)

I.e. we would separate the assertions with semi (which means we can't allow semi in extensions)

Reply:
We don't need or want multiple assertions, so let's not try.

Reply:
So, even if the user/PeerConnection provides multiple fingerprints (or multiple algorithms per provided fingerprint) to the IdP in the dictionary/hash, the IdP is always going to return a single assertion?

Reply:
Multiple fingerprints are included in the assertion.  The reason you might want to include multiple assertions are to support things like multiple IdPs or identities.  Right now, that's not a supported configuration.

Reply:
Whoops, yeah, that's a mistake.  I don't see how that would be possible.

Reply:
@cdh4u was this resolved?  I can't see that sentence so maybe it got removed?

Reply:
It has been removed. When I submitted the new version on the list, I did indicate that multiple identity assertions are not supported (they never were supported in the ABNF to begin with), and that anyone who has an issue with that should say so,